src/olympia/files/utils.py

```diff
@@ -398,8 +398,12 @@ def extract_search(content):
     rv = {}
     dom = minidom.parse(content)
 
-    def text(x):
-        return dom.getElementsByTagName(x)[0].childNodes[0].wholeText
+    def text(tag):
+        try:
+            return dom.getElementsByTagName(tag)[0].childNodes[0].wholeText
+        except (IndexError, AttributeError):
+            raise forms.ValidationError(
+                _('Could not parse uploaded file, missing <%s> element') % tag)
```

Comment:
An empty node is for us essentially missing so they're similar enough that I chose that wording. But yeah, maybe it's worth being even more explicit. I'll change that!